### PR TITLE
[v2] Topics/peer dependencies

### PR DESCRIPTION
**Pull request by flipactual**
per #1990

### CODE REVIEW

File: packages/gatsby-cli/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ package installed locally.
 
 ## Install
 
-`npm install --global gatsby-cli`
+`npm install --global gatsby-cli react react-dom`
```
Comment (pieh) [inline packages/gatsby-cli/README.md]: I don't think that's right. We don't want to install those globally.

Comment (flipactual) [reply packages/gatsby-cli/README.md]: womp, thank you

Comment (pieh) [reply packages/gatsby-cli/README.md]: I'm not expert on dependencies but these propably should be added to starters as dependencies but there are so many of those starters and many of them are unofficial so making react and react-dom peer dependencies will break them I think

Comment (flipactual) [reply packages/gatsby-cli/README.md]: I just arrived at the same conclusion 👍 

This is coming with a major version bump – I'd be surprised if this is the only thing that breaks compatibility with existing starters, but I don't know

**Comment by KyleAMathews** [comment]
Yeah, we'll have an upgrade guide which will say to install react/react-dom. Starters won't break because they'll have gatsby v1 in their package.json. But once their maintainers upgrade the starter, they'll add react/react-dom as dependencies so things will work fine.

File: packages/gatsby-cli/README.md (hunk 1 above)
Comment (KyleAMathews) [reply packages/gatsby-cli/README.md]: yeah — we'll upgrade the official starters and perhaps a few others. There'll be an upgrade guide for sites that starters will need to go through. It won't break anything though as a starter with a gatsby@1 dependency will keep working as normal.

**Pull request merged**